.github/workflows/multi_build.yml

```
@@ -0,0 +1,44 @@
+# Build is responsible for testing builds on all supported platforms.
+# It is broken up into three separate jobs with targeted builds so that each OS will
+# build in parallel and speed up overall CI time.
+name: Build
+on:
+  pull_request:
+
+jobs:
+  build_linux:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.17"
+          check-latest: true
+      - name: Build
+        run: make build-linux
```

It may be useful to add a cache for Go modules. I am not sure what the paths would be for darwin / windows.

```suggestion
      - name: Cache Go
        id: go-cache
        uses: actions/cache@v2
        env:
          cache-name: cache-go-modules
        with:
          path: |
            /home/runner/go/pkg/mod
            /home/runner/.cache/go-build
            /home/runner/go/bin
          key: v1-go-pkg-mod-${{ runner.os }}-${{ hashFiles('**/go.mod', '**/go.sum') }}

      - name: Build
        run: make build-linux
```

---

Cool looking into this.